integration/tests/cook/test_basic.py

```diff
@@ -760,3 +762,52 @@ def test_basic_docker_job(self):
         self.assertEqual(resp.status_code, 201)
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertEqual('success', job['instances'][0]['status'])
+
+    def get_executor(agent_state, instance):
+        for framework in agent_state['frameworks']:
+            for executor in framework['executors']:
+                if executor['id'] == instance['executor_id'][0]:
+                    return executor
+
+    def test_docker_port_mapping(self):
+        job_uuid, resp = util.submit_job(self.cook_url,
+                                         command='python -m http.server 8080',
+                                         ports=2,
+                                         container={'type': 'DOCKER',
+                                                    'docker': {'image': 'python:3.6',
+                                                               'network': 'BRIDGE',
+                                                               'port-mapping': [{'host-port': 0,
+                                                                                 'container-port': 8080},
+                                                                                {'host-port': 1,
+                                                                                 'container-port': 9090}]}})
+        self.assertEqual(resp.status_code, 201, resp.content)
+        try:
+            job = util.wait_for_job(self.cook_url, job_uuid, 'running')
+            instance = job['instances'][0]
+
+            # Get agent host/port
+            state = util.get_mesos_state(self.mesos_url)
+            agent = [agent for agent in state['slaves']
+                 if agent['hostname'] == instance['hostname']][0]
+            agent_hostport = agent['pid'].split('@')[1]
```

Comment:
I don't understand this line

Reply:
This is to get the host and port for the mesos agent. We already have the host but I wanted to avoid hardcoding the port, and this was the only place in the state I could find it. Added a comment.